web/server/codechecker_server/api/report_server.py

```diff
@@ -93,6 +91,10 @@ def verify_limit_range(limit):
     Max is returned if the value is larger than max.
     """
     max_query_limit = constants.MAX_QUERY_SIZE
+    # TODO why did we not check this before?
```

Comment:
I do not know why we didn't check it before. This week I run into this too because I tried to call the `getRunResults` function when limit parameter is null. I had ~60.000 reports and I used SQLite database and it tried to fetch all the reportrs with one query and my browser freezed.

We also use this function only in 3 places but we can use it in all places where a function can get a limit parameter.

Reply:
I've created a separate PR #2586 for this.